src/coreclr/nativeaot/Common/src/Internal/Runtime/CompilerHelpers/StartupCodeHelpers.cs

```diff
@@ -208,6 +208,27 @@ private static unsafe void RunEagerClassConstructors(IntPtr cctorTableStart, int
             }
         }
 
+        internal static unsafe void RunModuleInitializers()
```

Comment:
Would it be better to just emit the calls directly instead of going through the pointers?

Reply:
(It would be insufficient for multi-module mode, but the multi-module mode needs a more complex data structure to deal with the sorting problem anyway.)

Reply:
My line of thinking was that the current structure is closer to what we'll need for multi-module (and it actually works for multi-module, but with no ordering guarantees). If we emit the calls directly, it stops working for multi-module completely and cannot be extended to work for it.

Reply:
sounds reasonable